From 2541d4ee00ee2da0e8ee35f48ead7d1271cedc26 Mon Sep 17 00:00:00 2001
From: xdavidwu <xdavidwuph@gmail.com>
Date: Tue, 8 Aug 2023 00:11:58 +0800
Subject: [PATCH] meson: -lintl for musl+gettext

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 395eca1..c5cf936 100644
--- a/meson.build
+++ b/meson.build
@@ -322,7 +322,7 @@ conf.set10('FIRST_BOOT_FULL_PRESET',                          get_option('first-
 
 cc = meson.get_compiler('c')
 userspace_c_args = []
-userspace_c_ld_args = []
+userspace_c_ld_args = ['-lintl']
 meson_build_sh = find_program('tools/meson-build.sh')
 
 want_tests = get_option('tests')
-- 
2.41.0

