From 7c47949582cae5481a1b05c4ddc489e8fbb7a3b1 Mon Sep 17 00:00:00 2001
From: xdavidwu <xdavidwuph@gmail.com>
Date: Sat, 31 Dec 2022 23:21:53 +0800
Subject: [PATCH] bootctl: include bits/reg.h for musl __WORD_SIZE

---
 src/boot/bootctl.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/boot/bootctl.c b/src/boot/bootctl.c
index 7916587..7c5ae36 100644
--- a/src/boot/bootctl.c
+++ b/src/boot/bootctl.c
@@ -9,6 +9,7 @@
 #include <stdlib.h>
 #include <sys/mman.h>
 #include <unistd.h>
+#include <bits/reg.h>
 
 #include "sd-id128.h"
 
-- 
2.39.0

