From f8fc9170d42fce566071773e586d2e0985cc5d44 Mon Sep 17 00:00:00 2001
From: xdavidwu <xdavidwuph@gmail.com>
Date: Sat, 31 Dec 2022 23:42:41 +0800
Subject: [PATCH] errno-util: impl non-glibc STRERROR()

---
 src/basic/errno-util.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/basic/errno-util.h b/src/basic/errno-util.h
index 091f99c..112fd43 100644
--- a/src/basic/errno-util.h
+++ b/src/basic/errno-util.h
@@ -14,7 +14,15 @@
  * https://stackoverflow.com/questions/34880638/compound-literal-lifetime-and-if-blocks
  *
  * Note that we use the GNU variant of strerror_r() here. */
+#ifdef __GLIBC__
 #define STRERROR(errnum) strerror_r(abs(errnum), (char[ERRNO_BUF_LEN]){}, ERRNO_BUF_LEN)
+#else
+#define STRERROR(errnum) ({ \
+        char *m = alloca(ERRNO_BUF_LEN); \
+        strerror_r(abs(errnum), m, ERRNO_BUF_LEN); \
+        m; \
+})
+#endif
 
 /* A helper to print an error message or message for functions that return 0 on EOF.
  * Note that we can't use ({ … }) to define a temporary variable, so errnum is
-- 
2.39.0

